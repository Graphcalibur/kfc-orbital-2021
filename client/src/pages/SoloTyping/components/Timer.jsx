import React from 'react';

const Timer = (props) => {
    const { elapsed_time } = props;
    const padZeroes = (num) => num < 10 ? "0" + num.toString() : num;

    const mins = padZeroes(Math.floor(elapsed_time / 60000));
    const secs = padZeroes(Math.floor(elapsed_time / 1000) % 60);

    return <b>
        Time Taken: {mins}:{secs}
    </b>
}

export default Timer;