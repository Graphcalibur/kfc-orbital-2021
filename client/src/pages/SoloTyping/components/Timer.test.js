import React from "react";
import Timer from "./Timer";
import { create } from "react-test-renderer";

describe("Testing Timer component in page SoloTyping", () => {
  test("Test 1: Mins < 10, Seconds < 10, Not typing", () => {
    const tree = create(<Timer elapsed_time={1000} typing={false} />);
    expect(tree).toMatchSnapshot();
  });
  test("Test 2: Mins < 10, Seconds < 10, Typing", () => {
    const tree = create(<Timer elapsed_time={2000} typing={true} />);
    expect(tree).toMatchSnapshot();
  });
  test("Test 3: Mins < 10, Seconds >= 10, Not typing", () => {
    const tree = create(<Timer elapsed_time={12000} typing={false} />);
    expect(tree).toMatchSnapshot();
  });
  test("Test 4: Mins < 10, Seconds >= 10, Typing", () => {
    const tree = create(<Timer elapsed_time={12000} typing={true} />);
    expect(tree).toMatchSnapshot();
  });
  test("Test 5: Mins >= 10, Seconds < 10, Not typing", () => {
    const tree = create(<Timer elapsed_time={660000} typing={false} />);
    expect(tree).toMatchSnapshot();
  });
  test("Test 6: Mins >= 10, Seconds < 10, Typing", () => {
    const tree = create(<Timer elapsed_time={660000} typing={true} />);
    expect(tree).toMatchSnapshot();
  });
  test("Test 7: Mins >= 10, Seconds >= 10, Not typing", () => {
    const tree = create(<Timer elapsed_time={680000} typing={false} />);
    expect(tree).toMatchSnapshot();
  });
  test("Test 8: Mins >= 10, Seconds >= 10, Typing", () => {
    const tree = create(<Timer elapsed_time={680000} typing={false} />);
    expect(tree).toMatchSnapshot();
  });
});
