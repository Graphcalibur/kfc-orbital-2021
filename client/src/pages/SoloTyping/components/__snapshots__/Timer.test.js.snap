// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Timer component in page SoloTyping Test 1: Mins < 10, Seconds < 10, Not typing 1`] = `
<b
  className="text"
>
  Time Taken:
   
  <span
    style={
      Object {
        "color": "#AAAAAA",
      }
    }
  >
    00
    :
    01
  </span>
</b>
`;

exports[`Testing Timer component in page SoloTyping Test 2: Mins < 10, Seconds < 10, Typing 1`] = `
<b
  className="text"
>
  Time Taken:
   
  <span
    style={Object {}}
  >
    00
    :
    02
  </span>
</b>
`;

exports[`Testing Timer component in page SoloTyping Test 3: Mins < 10, Seconds >= 10, Not typing 1`] = `
<b
  className="text"
>
  Time Taken:
   
  <span
    style={
      Object {
        "color": "#AAAAAA",
      }
    }
  >
    00
    :
    12
  </span>
</b>
`;

exports[`Testing Timer component in page SoloTyping Test 4: Mins < 10, Seconds >= 10, Typing 1`] = `
<b
  className="text"
>
  Time Taken:
   
  <span
    style={Object {}}
  >
    00
    :
    12
  </span>
</b>
`;

exports[`Testing Timer component in page SoloTyping Test 5: Mins >= 10, Seconds < 10, Not typing 1`] = `
<b
  className="text"
>
  Time Taken:
   
  <span
    style={
      Object {
        "color": "#AAAAAA",
      }
    }
  >
    11
    :
    00
  </span>
</b>
`;

exports[`Testing Timer component in page SoloTyping Test 6: Mins >= 10, Seconds < 10, Typing 1`] = `
<b
  className="text"
>
  Time Taken:
   
  <span
    style={Object {}}
  >
    11
    :
    00
  </span>
</b>
`;

exports[`Testing Timer component in page SoloTyping Test 7: Mins >= 10, Seconds >= 10, Not typing 1`] = `
<b
  className="text"
>
  Time Taken:
   
  <span
    style={
      Object {
        "color": "#AAAAAA",
      }
    }
  >
    11
    :
    20
  </span>
</b>
`;

exports[`Testing Timer component in page SoloTyping Test 8: Mins >= 10, Seconds >= 10, Typing 1`] = `
<b
  className="text"
>
  Time Taken:
   
  <span
    style={
      Object {
        "color": "#AAAAAA",
      }
    }
  >
    11
    :
    20
  </span>
</b>
`;
